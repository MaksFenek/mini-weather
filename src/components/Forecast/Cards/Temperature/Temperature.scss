.temperature {
  margin-top: 24px;
  &-value {
    display: flex;

    h2 {
      margin-left: 24px;
      font-size: 2.5rem;
    }
  }

  h3 {
    font-size: 2rem;
    font-weight: normal;
    letter-spacing: 0.7px;
    margin: 12px 0;
  }
}
