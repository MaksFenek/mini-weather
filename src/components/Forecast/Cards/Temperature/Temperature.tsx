import React from 'react';
import './Temperature.scss';

interface ITemperature {
  temp: number | '--';
  description: string;
}

const Temperature: React.FC<ITemperature> = ({ temp, description }) => {
  return (
    <div className='forecast-content__today'>
      <div className='temperature'>
        <div className='temperature-value'>
          <svg className='icon' height='96px' width='96px'>
            <use xlinkHref={`icons/icons.svg#${description.toLowerCase()}`} />
          </svg>
          <h2>{temp} Â°C</h2>
        </div>
        <h3>{description}</h3>
      </div>
    </div>
  );
};

export default Temperature;
