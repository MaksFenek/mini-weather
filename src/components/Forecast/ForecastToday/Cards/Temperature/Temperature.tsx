// React
import React from 'react';

// Styles
import './Temperature.scss';

// Types
interface ITemperature {
  temp: number | '--';
  description: string;
}

const Temperature: React.FC<ITemperature> = ({ temp, description }) => {
  return (
    <div className='temperature'>
      <div className='temperature-value'>
        <svg className='icon' height='96px' width='96px'>
          <use
            aria-label='icon'
            xlinkHref={`icons/icons.svg#${description.toLowerCase()}`}
          />
        </svg>
        <h2 aria-label='temp'>{temp} Â°C</h2>
      </div>
      <h3 aria-label='temp'>{description}</h3>
    </div>
  );
};

export default Temperature;
