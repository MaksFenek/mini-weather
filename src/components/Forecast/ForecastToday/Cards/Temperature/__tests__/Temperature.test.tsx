import React from 'react';
import Temperature, { ITemperature } from '../Temperature';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

describe('<Temperature/>', () => {
  const props: ITemperature = {
    temp: 23,
    description: 'Clouds',
  };
  const wrapper = mount(<Temperature {...props} />);
  const tree = renderer.create(<Temperature {...props} />).toJSON();

  it('should render props correctly', () => {
    expect(wrapper.find(`[aria-label='temp']`).text()).toEqual(
      props.temp + ' Â°C'
    );
    expect(wrapper.find(`[aria-label='desc']`).text()).toEqual(
      props.description
    );
    expect(tree).toMatchSnapshot();
  });
});
