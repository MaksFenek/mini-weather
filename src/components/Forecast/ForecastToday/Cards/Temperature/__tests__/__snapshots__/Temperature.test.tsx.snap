// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Temperature/> should render props correctly 1`] = `
<div
  className="temperature"
>
  <div
    className="temperature-value"
  >
    <svg
      className="icon"
      height="96px"
      width="96px"
    >
      <use
        aria-label="icon"
        xlinkHref="icons/icons.svg#clouds"
      />
    </svg>
    <h2
      aria-label="temp"
    >
      23
       Â°C
    </h2>
  </div>
  <h3
    aria-label="desc"
  >
    Clouds
  </h3>
</div>
`;
